docs/sdk/transaction.md

```diff
@@ -137,6 +137,48 @@ const buildTxResult = await builder.buildDeployContractTx(
 console.log('unsigned transaction', buildTxResult.unsignedTx)
 ```
 
+## Gas Estimation
+
+On Alephium, each transaction involves two key parameters related to gas: `gasAmount` and `gasPrice`.
+
+- **`gasAmount`**: This represents the amount of gas required for on-chain computation and storage.
+- **`gasPrice`**: This is the amount of ALPH (Alephium's native token) that the user is willing to pay per unit of gas consumed during transaction execution.
+
+### Specifying `gasPrice`
+
+Users can set the `gasPrice` as long as it meets the [minimum required amount](/dapps/constants). 
+
+### Estimating `gasAmount`
+
+Estimating `gasAmount` is an integral part of building the transaction. The full node can accurately estimate the gas required based on the current state of the blockchain. However, this estimation can sometimes fail if the contract state changes due to other pending transactions in the mempool.
+
+### Using `gasEstimationMultiplier` (Optional)
+
+To mitigate the potential estimation errors for dApp transactions, a parameter called `gasEstimationMultiplier` can be used to scale up the estimated gas amount, providing a buffer to handle gas requirement changes.
+
+### Example
+
+Here’s how you can use extract the estimated `gasAmount` and optionally use `gasEstimationMultiplier` in your code:
```

Comment:
[Nit] Do we wanna mention that the value of `gasEstimationMultiplier` is between `1` and `2`?